### PR TITLE
[samplecode][1/3]Implement Stream.Server sample code

### PR DESCRIPTION
**Pull request by summer-ji-eng**
Stream.Server Sample Code

```
   * try (EchoClient echoClient = EchoClient.create()) {
   *   ExpandRequest request = ExpandRequest.newBuilder().build();
   *   ServerStream&lt;EchoResponse&gt; stream = echoClient.expandCallable().call(request);
   *   for (EchoResponse response : stream) {
   *     // Do something when receive a response
   *   }
   * }
```
Sister PR #631

### CODE REVIEW

File: src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java [hunk 1]
```diff
@@ -783,6 +783,8 @@ private static MethodDefinition createCallableMethod(
             .build();
 
     Optional<String> sampleCode = Optional.empty();
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java:785]: While we're here, could we rename this to `sampleCodeOpt`?

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 2]
```diff
@@ -512,6 +513,44 @@ public static String composePagedCallableMethodHeaderSampleCode(
...
+            Variable.builder()
+                .setName(JavaStyle.toLowerCamelCase(clientType.reference().name()))
+                .setType(clientType)
+                .build());
+    // Assign method's request variable with the default value.
+    VariableExpr requestVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder().setName("request").setType(method.inputType()).build());
+    Message requestMessage = messageTypes.get(method.inputType().reference().simpleName());
+    Preconditions.checkNotNull(requestMessage);
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:532]: Add an error message?

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 3]
```diff
@@ -647,6 +686,68 @@ public static String composePagedCallableMethodHeaderSampleCode(
     return bodyExprs.stream().map(e -> ExprStatement.withExpr(e)).collect(Collectors.toList());
   }
 
+  private static List<Statement> composeStreamServerSampleCodeBodyStatements(
+      Method method, VariableExpr clientVarExpr, AssignmentExpr requestAssignmentExpr) {
+    List<Expr> bodyExprs = new ArrayList<>();
+    bodyExprs.add(requestAssignmentExpr);
+
+    // Create ServerStream variable expression, and assign it with invoking server stream method.
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:694]: Nit: s/with/by

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 4]
```diff
@@ -647,6 +686,68 @@ public static String composePagedCallableMethodHeaderSampleCode(
...
+            .setIsDecl(true)
+            .build();
+    ForStatement forStatement =
+        ForStatement.builder()
+            .setLocalVariableExpr(responseVarExpr)
+            .setCollectionExpr(serverStreamVarExpr)
+            .setBody(
+                Arrays.asList(
+                    CommentStatement.withComment(
+                        LineComment.withComment("Do something when receive a response."))))
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:744]: Nit: "Do something when a response is received." or similar.

**Review by miraleung — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java (hunk 1 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java:785]: Sure.

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 2 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:532]: Sure, updated in other 4 places in this files.

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 3 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:694]: `// Create server stream variable expression, and assign it a value by invoking server stream method.` ?

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 4 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java]: Done

**Review by xiaozhenliu-gg5 — APPROVED** [review]

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 5]
```diff
@@ -253,7 +254,11 @@ public static String composeRpcDefaultMethodHeaderSampleCode(
             Variable.builder().setName("request").setType(method.inputType()).build());
     List<VariableExpr> rpcMethodArgVarExprs = Arrays.asList(requestVarExpr);
     Message requestMessage = messageTypes.get(method.inputType().reference().simpleName());
-    Preconditions.checkNotNull(requestMessage);
+    Preconditions.checkNotNull(
+        requestMessage,
+        String.format(
+            "The request `%s` is not defined as message type.",
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:260]: Could we please make the error message more accurate? E.g. "Could not find the message type %s".

**Review by miraleung — APPROVED** [review]

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 5 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:260]: Will update the error message.

**Pull request merged**